x/sequencer/types/metadata.go

```diff
@@ -33,13 +40,40 @@ func (d SequencerMetadata) Validate() error {
 		return errorsmod.Wrap(err, "invalid rest api URLs")
 	}
 
+	if d.FeeDenom != nil {
+		if err := d.FeeDenom.Validate(); err != nil {
+			return errors.Join(ErrInvalidFeeDenom, err)
+		}
+	}
+
 	if d.ContactDetails == nil {
 		return nil
 	}
 
 	return d.ContactDetails.Validate()
 }
 
+func (dm DenomMetadata) IsSet() bool {
+	return dm != DenomMetadata{}
+}
+
+func (dm DenomMetadata) Validate() error {
+	if err := sdk.ValidateDenom(dm.Base); err != nil {
+		return fmt.Errorf("invalid metadata base denom: %w", err)
+	}
+
+	if err := sdk.ValidateDenom(dm.Display); err != nil {
+		return fmt.Errorf("invalid metadata display denom: %w", err)
+	}
+
+	// validate exponent
+	if AllowedDecimals(dm.Exponent) != Decimals18 {
+		return fmt.Errorf("invalid exponent")
+	}
+
+	return nil
+}
```

Comment:
pretty sure we have similar code in several other places but nvm

Reply:
@danwt no, we have exactly the same code. But copying is less risky right now